test/e2e/parametrisable_templates/parametrisable_templates_test.go

```diff
@@ -0,0 +1,440 @@
+			response, err = harness.Client.ReadDeviceWithResponse(harness.Context, deviceId)
+			Expect(err).ToNot(HaveOccurred())
+			device = response.JSON200
+			Expect(device).ToNot(BeNil(), "failed to read updated device")
+			inlineConfigPathResponse, err := harness.GetDeviceInlineConfig(device, inlineConfigName)
+			Expect(err).ToNot(HaveOccurred())
+			if len(inlineConfigPathResponse.Inline) > 0 {
+				Expect(inlineConfigPathResponse.Inline[0].Path).To(ContainSubstring(teamLabelValue))
+				Expect(inlineConfigPathResponse.Inline[0].Content).To(ContainSubstring(defaultTeamLabelValue))
+			}
+		})
+
+		It(`Verifies that if a device is missing a parametrisable device label
+		    an error is generated, but it will reconcile if the label is provided`,
+			Label("75600"), func() {
+
+				By("Check the device status is Online")
+				CheckDeviceOnlineStatus(harness, deviceId)
+
+				By("Create a fleet with a template variable")
+				err := configProviderSpec.FromInlineConfigProviderSpec(inlineConfigValidWithFunction)
+				Expect(err).ToNot(HaveOccurred())
+				err = harness.CreateTestFleetWithConfig(fleetTestName, testFleetSelector, configProviderSpec)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Associate the device to the fleet without adding the team label")
+				nextRenderedVersion, err := harness.PrepareNextDeviceVersion(deviceId)
+				Expect(err).ToNot(HaveOccurred())
+
+				harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+
+					(*device.Metadata.Labels)[fleetSelectorKey] = fleetSelectorValue
+					logrus.Infof("Updating %s with label %s=%s", deviceId,
+						fleetSelectorKey, fleetSelectorValue)
+				})
+
+				By("Check the device will fail to reconcile")
+				harness.WaitForDeviceContents(deviceId, `The device could not be updated to the fleet`,
+					func(device *v1alpha1.Device) bool {
+						return device.Status.Updated.Status == v1alpha1.DeviceUpdatedStatusOutOfDate
+					}, "2m")
+				resp, err := harness.Client.ReadDeviceStatusWithResponse(harness.Context, deviceId)
+				Expect(err).ToNot(HaveOccurred())
+				device := resp.JSON200
+				Expect((*device.Metadata.Annotations)["fleet-controller/lastRolloutError"]).NotTo(BeNil())
+				Expect(device.Status.Updated.Status).To(Equal(v1alpha1.DeviceUpdatedStatusOutOfDate))
+				Expect((*device.Metadata.Annotations)["fleet-controller/lastRolloutError"]).To(ContainSubstring("no entry for key \"team\""))
+
+				By("Add the team label to the device")
+				harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+
+					(*device.Metadata.Labels)[teamLabelKey] = teamLabelValue
+					logrus.Infof("Updating %s with label %s=%s", deviceId,
+						teamLabelKey, teamLabelValue)
+				})
+				resp, err = harness.Client.ReadDeviceStatusWithResponse(harness.Context, deviceId)
+				Expect(err).ToNot(HaveOccurred())
+				device = resp.JSON200
+				Expect((*device.Metadata.Labels)[teamLabelKey]).To(ContainSubstring(teamLabelValue))
```

Comment:
GetDevice

Although, checking that a label you set is there... should not be part of this test (seeing it in some other places) that's something that should always work. 

You may want to test it, of course, but on an independent test may be, targeted specifically tho this. This would save a few 20-30 lines on the tests here :)